tasks/phpstan/phpstan.neon

```diff
@@ -4,3 +4,5 @@ parameters:
 	ignoreErrors:
 		- '#Function members_register_.+ not found#'
 		- '#MailPoet\\Premium\\DI\\ContainerConfigurator not found#' # this class is not available when premium is not active
+		- '#Class MailPoet\\Premium\\DI\\ContainerConfigurator does not have a constructor#'
```

Comment:
I'm just curious... how this can happen? Do we ever try to pass some parameters to ContainerConfigurator instantiation?

Reply:
Another good catch. Looks like I didn't inspect it attentively enough.

https://github.com/mailpoet/mailpoet/blob/master/lib/DI/ContainerWrapper.php#L64
Here `$free_container` is being passed in as an argument to a `ContainerConfigurator` instance.
Looks like it is not necessary, as two lines below it sets `$free_container` on a `$premium_container` instance.

I think it's safe to fix the rule by removing the unnecessary `ContainerWrapper` constructor argument, and removing the `ignoreErrors` rule.

Reply:
Then it is PHPStan's good catch :)

Reply:
Hey! Good catch! It must be some left over after a refactor.